pkg/helper/client_test.go

```
@@ -0,0 +1,117 @@
+- name: eth1
+  state: up
+  type: ethernet
+- name: vethab6030bd
+  state: down
+  type: ethernet
+`)
+			filteredState = nmstatev1alpha1.State(`interfaces:
+- name: eth1
+  state: up
+  type: ethernet
+`)
+			interfacesFilterGlob = glob.MustCompile("veth*")
+		})
+
+		It("should filter out matching interface and keep the others", func() {
+			returnedState, err := filterOut(state, interfacesFilterGlob)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedState).To(Equal(filteredState))
+		})
+	})
+
+	Context("when the filter is matching multiple interfaces in the list", func() {
+		BeforeEach(func() {
+			state = nmstatev1alpha1.State(`interfaces:
+- name: eth1
+  state: up
+  type: ethernet
+- name: vethab6030bd
+  state: down
+  type: ethernet
+- name: vethjyuftrgv
+  state: down
+  type: ethernet
+`)
+			filteredState = nmstatev1alpha1.State(`interfaces:
+- name: eth1
+  state: up
+  type: ethernet
+`)
+			interfacesFilterGlob = glob.MustCompile("veth*")
+		})
+
+		It("should filter out all matching interfaces and keep the others", func() {
+			returnedState, err := filterOut(state, interfacesFilterGlob)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(returnedState).To(Equal(filteredState))
+		})
+	})
+
+	Context("when the filter is matching multiple prefixes", func() {
+		BeforeEach(func() {
+			state = nmstatev1alpha1.State(`interfaces:
+- name: eth1
+  state: up
+  type: ethernet
+- name: vethab6030bd
+  state: down
+  type: ethernet
+- name: vnet2b730a2b@if3
+  state: down
+  type: ethernet
+`)
+			filteredState = nmstatev1alpha1.State(`interfaces:
+- name: eth1
+  state: up
+  type: ethernet
+`)
+			interfacesFilterGlob = glob.MustCompile("{veth*,vnet*}")
+		})
+
+		It("it should filter out all interfaces matching any of these prefixes and keep the others", func() {
+			returnedState, err := filterOut(state, interfacesFilterGlob)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(returnedState).To(Equal(filteredState))
+		})
+	})
```

Can we add a bad filter expression test ? or this concept does not exist at globg ?

---

Filter our can be broken by used only when they set invalid glob. That would however fail early in init(), not as a part of filterOut. I think we cannot test it (and there is no real need, as it would fail to run the handler completely).